### ISSUE TITLE
usersettings -> personal does not validate

### ISSUE THREAD

**Issue by RogerHaase** [moinwiki/moin#855]
After changing fields and clicking save a "Nothing changed" message is flashed and disappears after view is reset to original values.

No problems noted on other usersettings forms,

**Comment by ThomasWaldmann** [comment]
I had a look and found that the "personal" form does not validate, no matter what you have in there / have changed there.

Did this work with moin2 on python2?

**Comment by RogerHaase** [comment]
I think the bug in in flatland/schema/containters.py, _set_flat method near line 839. None values are returned for the timezone and locale fields causing validation to fail because they are required fields.

**Comment by ThomasWaldmann** [comment]
I found it, flatland bug.

```
    # flatland.schema.base:641 has a bug
    def set_flat(self, pairs, sep=u'_'):
        self.raw = Unset
        if hasattr(pairs, 'items'):
            pairs = pairs.items()

        return self._set_flat(list(pairs), sep)  # XXX added list() here
```

The problem is that if pairs is a generator, it might be fully iterated multiple times (and be exhausted after first full iteration) by _set_flat().

**Comment by ThomasWaldmann** [comment]
https://github.com/discorporate/flatland/pull/54

**Comment by RogerHaase** [comment]
No bug is in usersettings forms. The personal form has 2 fields named 'name'. 

Yes, that this worked in py2, amazing?

**Comment by RogerHaase** [comment]
haha, two birds with one stone

**Comment by ThomasWaldmann** [comment]
Should be fixed now with the flatland 0.9.1 release.
